netperf_tool/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+cmake_minimum_required(VERSION 3.5)
+project(netperf_tool)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  # About -Wno-sign-conversion: With Clang, -Wconversion implies -Wsign-conversion. There are a number of
+  # implicit sign conversions in rclcpp and gtest.cc, see https://ci.ros2.org/job/ci_osx/9265/.
+  # Hence disabling -Wsign-conversion for now until all those have eventually been fixed.
+  # (from https://github.com/ros2/rclcpp/pull/1188#issuecomment-650229140)
+  add_compile_options(-Wall -Wextra -Wconversion -Wno-sign-conversion -Wpedantic -Wnon-virtual-dtor -Woverloaded-virtual)
+endif()
+
+# Figure out Python3 debug/release before anything else can find_package it
+if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+  find_package(python_cmake_module REQUIRED)
+  find_package(PythonExtra REQUIRED)
+
+  # Force FindPython3 to use the debug interpreter where ROS 2 expects it
+  set(Python3_EXECUTABLE "${PYTHON_EXECUTABLE_DEBUG}")
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_python REQUIRED)
+_ament_cmake_python_get_python_install_dir()
+find_package(ament_cmake_ros REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(netperf_tool_interfaces REQUIRED)
+
+# Find python before pybind11
+find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+
+find_package(pybind11_vendor REQUIRED)
+find_package(pybind11 REQUIRED)
```

Comment:
It seems odd to find Python two different ways. I'm not sure if this is the best practice or not?

Reply:
I think that we're doing this in other places as well, if not it breaks or Windows debug builds.
If all our platforms would use cmake 3.16 or later, we could use `Python3_FIND_ABI` from [FindPython3](https://cmake.org/cmake/help/latest/module/FindPython3.html) which should work.
Before that version, I think there wasn't a way to make `FindPython3` find the python debug version when both ABIs were available, except for already specifying `Python3_EXECUTABLE` beforehand.

According to rep2000, all our platforms already support cmake 3.16, except for macOS (which uses 3.14.4).
If we don't mind breaking that, I will try using the `Python3_FIND_ABI` hint.

Reply:
Are we still supporting macOS?

Reply:
>  Either way, the cmake version isn't an issue for macOS.

I didn't know that.
I guess that rep 2000 is indicating that because that was the version of cmake we used to use in macOS CI.
But we're not running that anymore, so I think that requiring a newer minimal cmake version it's worth it.
I think that I'm going to do that in a different PR though, as if not I will block this on getting windows debug builds working (which is always not completely straightforward).

Reply:
> I guess that rep 2000 is indicating that because that was the version of cmake we used to use in macOS CI.

I think that's right.

Can you bring this up at the next ROS 2 meeting? It sounds like it would be an improvement for all of our packages that use Python.

On VIPER we run into the issue that finding python in two ways caused us to find two different python versions in a single build, which broke it.

Reply:
> On VIPER we run into the issue that finding python in two ways caused us to find two different python versions in a single build, which broke it.

Oh, that's interesting ...
Sure, I can bring that up